crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -77,10 +81,19 @@ impl Actor for SoftwareManagerActor {
         let operation_logs = OperationLogs::try_new(self.config.log_dir.clone().into())
             .map_err(SoftwareManagerError::FromOperationsLogs)?;
 
+        let sudo: Option<PathBuf> = match Command::new(SUDO)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+        {
+            Err(err) if err.kind() == NotFound => None,
```

Comment:
The same as in `restart_manager/actor.rs`. The next conditional arm includes the case when command sudo returns an error except NotFound. But the following code uses sudo, which sounds not correct to me.

I would remove if err.kind() == ErrorKind::NotFound and just not using sudo if any error occurs after execute sudo.